src/AWSS3.jl

```diff
@@ -106,7 +107,7 @@ function s3(aws::AWSConfig,
         end
         request[:url] = url
 
-        response = AWSCore.do_request(request)
+        response = AWSCore.do_request(request; return_headers=return_headers)
```

Comment:
This could return a tuple, which means you need to change the `isa` check below.

Maybe `(response isa XMLDict.XMLDictElement || response isa Tuple{XMLDict.XMLDictElement, Any})`?

Reply:
Could also do:

```julia
response = AWSCore.do_request(request; return_headers=return_headers)

if response isa Tuple{XMLDict.XMLDictElement, Any}
  response = response[1]
end

if response isa XMLDict.XMLDictElement && ...
```

or this way, which I like a lot less

```julia
if return_headers
  response, _ = AWSCore.do_request(request; return_headers=return_headers)
else
  response = AWSCore.do_request(request; return_headers=return_headers)
end

if response isa XMLDict.XMLDictElement && ...
```

Reply:
Well the first way discards the headers, but you need to return them later if `return_headers`